ext/js/lib/js.rb

```diff
@@ -132,6 +149,55 @@ def await
     promise = JS.global[:Promise].resolve(self)
     JS.promise_scheduler.await(promise)
   end
+
+  # Return string that can be embedded in a JavaScript statement
+  def to_js_string
+    case self.typeof
+    when "string"
+      return "\"#{self}\""
+    when "object"
+      return JS.global[:JSON].stringify(self).to_s
+    else
+      self.to_s
+    end
+  end
+end
+
+# A wrapper class for creating JavaScript Arrays in Ruby.
+class JS::Array
```

Comment:
I don't want to define this kind of wrapper class by design to avoid complex type-conversion round trips.

Can we define `def to_js: () -> JS::Object` under `Array` instead of having Array specific wrapper like below?

```ruby
class Array
  def to_js
    new_array = JS.eval("return []")
    self.each do |element|
      # NOTE: This JS method call implicitly convert element to JS object by to_js
      new_array.push element
    end
    new_array
  end
end
```

Reply:
I didn't come up with the idea to implement Array#to_js.
Once I implemented Array#to_js, it became even simpler. Thank you very much.

I also implemented Hash#to_js and NilClass#to_js. 
Please point out if the location or name of the files is inappropriate.